File: Leetcode75-2024/remove-stars.java
import java.util.*;

class Solution {
    public String removeStars(String s) {
        
        Stack<Character> stack = new Stack<>();
        int index = 0;
        int lengthS = s.length();
        for(index=0; index < lengthS ; index++){
            char str = s.charAt(index);

            stack.push(str);

            if(str == '*'){
                stack.pop();
                stack.pop();
            }
        }

        StringBuilder finalStr = new StringBuilder();
        for (char c : stack) {
            finalStr.append(c);
        }
        
        return finalStr.toString();
    }
}

    

File: Leetcode75-2024/remove-stars.py
class Solution(object):
    def removeStars(self, s):
        """
        :type s: str
        :rtype: str
        """
        
        stack = []
        for i in s:
            stack.append(i)
            if i == "*":
                stack.pop()
                stack.pop()
        
        return ''.join(stack)